snuba/query/expressions.py

```diff
@@ -319,8 +340,15 @@ def transform(self, func: Callable[[Expression], Expression]) -> Expression:
         Applies the transformation to the inner expression but not to the parameters
         declaration.
         """
-        transformed = replace(self, transformation=self.transformation.transform(func))
-        return func(transformed)
+        transformed_expression = self.transformation.transform(func)
+        if transformed_expression != self.transformation:
+            return func(
+                replace(self, transformation=self.transformation.transform(func))
+            )
+        else:
+            # Does instantiate a copy of the this class if the children did
```

Comment:
This comment seems counter to the tests. Here you say a copy is created but the test asserts that a new copy is not created.

Reply:
I forgot a `not`